File: Maximum Sum LCM/code.java
class Solution {
    static long maxSumLCM(int n) {
        // code here
        if(n==1){
            return 1;
        }
        long sum=0;
        for(int i=1; i*i<=n; i++){
            if(n%i==0){
                if(n/i==i){
                    sum+=i;
                }
                else{
                sum+=i;
                sum+=n/i;
                }
            }
        }
        return sum;
    }
}

File: Minimize the sum/code.cpp
class Solution {
public:
    int minimizeSum(int N, vector<int> arr) {
        // code here
        priority_queue<int, vector<int>, greater<int>> h;
        
        for(int i = 0; i < N; i++){
            h.push(arr[i]);
        }
        
        int ans = 0;
        while(h.size() > 1){
            int sum = h.top();
            h.pop();
            sum += h.top();
            h.pop();
            h.push(sum);
            ans += sum;
        }
        
        return ans;
        
    }
};
